pkg/bisect/bisect_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2019 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package bisect
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/google/syzkaller/pkg/instance"
+	"github.com/google/syzkaller/pkg/mgrconfig"
+	"github.com/google/syzkaller/pkg/report"
+	"github.com/google/syzkaller/pkg/vcs"
+)
+
+// testEnv will implement instance.BuilderTester. This allows us to
+// set bisect.env.inst to a testEnv object.
+type testEnv struct {
+	repo *vcs.TestRepo
+	r    vcs.Repo
+	t    *testing.T
+	// TODO: add a "fix bool" here so that Test() can return results according to
+	// whether fix/cause bisection is happening.
+}
+
+func (env *testEnv) BuildSyzkaller(repo, commit string) error {
+	return nil
+}
+
+func (env *testEnv) BuildKernel(compilerBin, userspaceDir, cmdlineFile, sysctlFile string,
+	kernelConfig []byte) (string, error) {
+	return "", nil
+}
+
+func (env *testEnv) Test(numVMs int, reproSyz, reproOpts, reproC []byte) ([]error, error) {
+	hc, err := env.r.HeadCommit()
+	if err != nil {
+		env.t.Fatal(err)
+	}
+	parts := strings.Split(hc.Title, ".")
+	major, err := strconv.Atoi(parts[0])
+	if err != nil {
+		env.t.Fatalf("invalid test commit msg '%v'", hc.Title)
+	}
+	minor, err := strconv.Atoi(parts[1])
+	if err != nil {
+		env.t.Fatalf("invalid test commit msg '%v'", hc.Title)
+	}
+	// For cause bisection, if newer than or equal to 6.2, it crashes.
+	// -- 6.2 is the cause commit.
+	// TODO: for fix bisection(check env.fix), if older than 6.2, it crashes.
+	// -- 6.2 is the fix commit.
+	if major == 6 && minor >= 2 || major > 6 {
```

Comment:
I would name commits as `fmt.Sprint(rv * 100 + i)` and then all of this will become much simpler. This check will be `commit >= 602`. Later we will need to specify this breakage point in the tests and probably also some broken commits that need to be skipped. Specifying them as a single number will make things easier as well.

Reply:
Thank you! I've made the changes and will upload the newer PR.